### PR TITLE
Cql3: lift infinite bound check

### LINKED ISSUE #432: Support range deletions in CQL
from https://issues.apache.org/jira/browse/CASSANDRA-6237

given a table like:

```
CREATE TABLE events (
    id text,
    created_at timestamp,
    content text,
    PRIMARY KEY (id, created_at)
)
```

allow queries like:

```
DELETE FROM events WHERE id='someEvent' AND created_at < 'Jan 3, 2013';
```

### PR DESCRIPTION
**Pull request by bhalevy**
If the database supports infinite bound range deletions,
CQL layer will no longer throw an error indicating that both ranges
need to be specified.
    
Fixes #432
    
(cherry picked from commit 17acedb0c610d69cbc9675a02af467e44fa5b175)
    
Update test_range_deletion_scenarios unit test accordingly.

### CODE REVIEW

File: tests/cql_query_test.cc [hunk 1]
```diff
@@ -1064,18 +1064,12 @@ SEASTAR_TEST_CASE(test_range_deletion_scenarios) {
...
-        try {
-            e.execute_cql("delete from cf where p = 1 and c <= 3").get();
-            BOOST_FAIL("should've thrown");
-        } catch (...) { }
-        try {
-            e.execute_cql("delete from cf where p = 1 and c >= 0").get();
-            BOOST_FAIL("should've thrown");
-        } catch (...) { }
+        e.execute_cql("delete from cf where p = 1 and c <= 3").get();
+        e.execute_cql("delete from cf where p = 1 and c >= 8").get();
```
Comment (dekimir) [inline tests/cql_query_test.cc]: I don't understand why this replacement code (as opposed to just deleting lines 1067--1074).  The CQL in those statements was not meant to be executed.

Also, there should still be a test for rejecting infinite-range deletions when they happen to be prohibited.

**Comment by avikivity** [comment]
@bhalevy why did you merge this? You are not a maintainer.

**Comment by avikivity** [comment]
dequeued, merged out of process

**Comment by bhalevy** [comment]
> @bhalevy why did you merge this? You are not a maintainer.

Sorry, you're right. I got used to doing that following the pull request process in other repos.

**Comment by bhalevy** [comment]
I wonder if there's a way to reopen the PR...

**Comment by avikivity** [comment]
@bhalevy please drop the cherry-picked-from markers (they refer to non-existent commits) and repush.

**Comment by bhalevy** [comment]
@avikivity done.
Tree: https://github.com/bhalevy/scylla/tree/cql3-lift-infinite-bound-check
Commit: add40d4e59861b96295c8a3d2245fd84b8ab4b28

**Comment by avikivity** [comment]
Queued as d47a07814bc47a8cdbbc96ddf586b7da8cb27c56.

**Pull request merged**